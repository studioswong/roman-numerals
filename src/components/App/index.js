import React, { Component } from 'react';
import styled from 'styled-components';
import logo from './logo.svg';
import Form from '../Form';
const FormDiv = styled.div`
  padding: 16px;
`;

const AppDiv = styled.div`
  text-align: center;
`;

const Logo = styled.img`
  height: 120px;
`;

class App extends Component {
  render() {
    return (
      <AppDiv>
        <header>
          <Logo src={logo} alt="logo" />
          <h1>Roman Numerals Convertor</h1>
        </header>
        <FormDiv><Form /></FormDiv>
      </AppDiv>
    );
  }
}

export default App;
