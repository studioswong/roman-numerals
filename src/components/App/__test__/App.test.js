/**
 * @jest-environment node
 */

import React from 'react';
import { shallow } from 'enzyme';
import App from '../';

describe('App UI', () => {
    it('should render', () => {
        const wrapper = shallow(<App />);
        expect(wrapper).toMatchSnapshot();
    });
});
