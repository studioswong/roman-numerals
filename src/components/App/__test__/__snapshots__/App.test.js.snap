// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App UI should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <header>
          <h1>
            Roman Numerals Convertor
          </h1>
        </header>,
        <styled.div>
          <Form />
        </styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Roman Numerals Convertor
          </h1>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Roman Numerals Convertor",
          },
          "ref": null,
          "rendered": "Roman Numerals Convertor",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Form />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <header>
            <h1>
              Roman Numerals Convertor
            </h1>
          </header>,
          <styled.div>
            <Form />
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Roman Numerals Convertor
            </h1>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Roman Numerals Convertor",
            },
            "ref": null,
            "rendered": "Roman Numerals Convertor",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Form />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
